src/plugins/discover/public/application/angular/discover.js

```diff
@@ -723,11 +727,6 @@ function discoverController($route, $scope, Promise) {
       visStateAggs
     );
 
-    $scope.volatileSearchSource.onRequestStart((searchSource, options) => {
-      if (!$scope.opts.chartAggConfigs) return;
-      return $scope.opts.chartAggConfigs.onSearchRequestStart(searchSource, options);
-    });
-
```

Comment:
Dear @nreese, asking you because a long time ago (#14157) you've introduced `onSearchRequestStart` here to recalculate min, max of the histogram aggregation. Pls correct me if I'm wrong, but I don't think it's necessary here? Made this PR to test if this leads to a test failure, if it's not needed I'd like to remove it because `$scope.volatileSearchSource.onRequestStart` adds one more execution of this function every time data is fetched. can be a lot after a while.

Reply:
onSearchRequestStart for histograms was added to avoid cases where the requested number of aggregation buckets would be so large as to cause rendering problems. A preflight request was needed to get the min and max so a proper scale could be passed to the histogram aggregation request. There has been a lot of refactored code since that PR merged so its possible this is no longer needed in discover since the time min and max are known from the global time picker. Does discover properly bound the number of buckets based on the global time range?

Reply:
@nreese thx for the context, I think the right bucket size is chosen here:

https://github.com/elastic/kibana/blob/c33987d92977bab6e9a1eeffd49985297ca72cf1/src/plugins/data/common/search/aggs/buckets/date_histogram.ts#L187-L200

Reply:
Using `debugger` I couldn't find code that would trigger a pre flight request, should be safe to remove